gateway-service/src/main/java/org/zowe/apiml/gateway/filters/pre/JWTAuthenticationFilter.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.gateway.filters.pre;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.AuthenticationFailureHandler;
+import org.zowe.apiml.gateway.security.service.schema.source.AuthSource;
+import org.zowe.apiml.gateway.security.service.schema.source.JwtAuthSource;
+import org.zowe.apiml.gateway.security.service.schema.source.JwtAuthSourceService;
+import org.zowe.apiml.security.common.login.NonCompulsoryAuthenticationProcessingFilter;
+import org.zowe.apiml.security.common.token.TokenAuthentication;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Optional;
+
+@Slf4j
+public class JWTAuthenticationFilter extends NonCompulsoryAuthenticationProcessingFilter {
+    private final AuthenticationFailureHandler failureHandler;
+
+    private final JwtAuthSourceService jwtAuthSourceService;
+
+       public JWTAuthenticationFilter(String endpoint,
+                                    AuthenticationFailureHandler failureHandler,
+                                      JwtAuthSourceService jwtAuthSourceService) {
+        super(endpoint);
+        this.failureHandler = failureHandler;
+        this.jwtAuthSourceService = jwtAuthSourceService;
+    }
+        @Override
+        public Authentication attemptAuthentication(HttpServletRequest request, HttpServletResponse response) {
+            Optional<String> jwtToken = jwtAuthSourceService.getToken(request);
+
+
+            if (jwtToken.isPresent()) {
+                log.debug("jwt access token found in the request");
+                AuthSource authSource = new JwtAuthSource(jwtToken.get());
+
+                AuthSource.Parsed parsedToken = jwtAuthSourceService.parse(authSource);
+                if (parsedToken != null && StringUtils.isNotEmpty(parsedToken.getUserId())) {
+                    return TokenAuthentication.createAuthenticated(parsedToken.getUserId(), jwtToken.get());
+                }
+            }
+
+            log.debug("No valid jwt access token found in the request.");
```

Comment:
Same as OIDC, should it be error?

Reply:
Same as OIDC, I don't think we need this message as error.